### ISSUE TITLE
[FEATURE] <VPN-Verbindung IOS--warten bis VPN aufgebaut bei App Start>

### ISSUE THREAD

**Issue by Uwe1984** [foxriver76/ioBroker-Visu-App#35]
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Feature description

Wie im ioBroker Forum beschrieben, habe ich eine Möglichkeit gefunden, eine VPN Verbindung bei App Start automatisiert aufzubauen. ( Gilt nur für Apple-IOS )
Leider dauert das aufbauen der VPN Verbindung ca. 2-3 s und die App wirft dann eine Fehlermeldung aus. 
Es wäre super, wenn es eventuell eine Einstellungsmöglichkeit gäbe, um den App Start etwas zu verzögern, damit Zeit für die VPN Verbindung besteht. 

Ich bin der User K_o_bold aus dem Forum. 
Gerne kann ich nochmal meine Beobachtungen, wie im Forum geschildert beschreiben, falls gewünscht.

**Comment by Uwe1984** [comment]
Ich bin nicht im "TestFlight"... Ich kann aber gerne beim Testen unterstützen. Wenn du mir sagst was ich tun muss, gerne.

**Comment by foxriver76** [comment]
https://testflight.apple.com/join/UWrb8L7e

Dann hast du mit der offiziellen TestFlight App Zugriff auf die aktuelle Beta Version

**Comment by Uwe1984** [comment]
Ok die App habe ich geladen. Sag bescheid wenn du soweit bist

**Comment by foxriver76** [comment]
Test ist aktiv

**Comment by foxriver76** [comment]
Hat sich das Verhalten für dich gebessert @Uwe1984?

**Comment by Uwe1984** [comment]
Nein, bisher nicht. Ich habe auf Testflight alle neuen Versionen getestet und dir auch auf diesem Weg eine Rückmeldung gegeben. 
Wenn ich noch was tun kann, oder weitere Infos benötigt werden, dann gerne

**Comment by foxriver76** [comment]
Puh okay also zumindest bei mir mit Auto VPN Pro + lokal geht er reibungslos in den VPN falls verfügbar

**Comment by foxriver76** [comment]
Probier mal extern mit url gleiche wie intern und vpn Erkennung + Auto VPN.

**Comment by Uwe1984** [comment]
Ja so funktioniert es jetzt auch bei mir...Ich habe es jetzt bestimmt 10 mal getestet. 
Was ein bisschen unschön ist, die Fehlermeldung wie oben beschrieben kommt trotzdem kurz. 

Das ich die Einstellungen so wählen muss, ist aber  nicht gerade selbsterklärend

**Comment by Uwe1984** [comment]
Was mir auch gerade auffällt, wenn ich im Wlan bin, dann öffnet sich die Vis sehr schnell, allerdings schaltet sich das VPN ja trotzdem ein und dann lädt die Vis neu. 
Ist das bei dir auch so?

**Comment by foxriver76** [comment]
Nein, hast du deine WLAN SSID konfiguriert? Die App kennt zwei Stati lokal und extern, lokal immer wenn VPN oder mit der SSID verbunden. 

Und das ist auch der Hintergrund aktuell der Konfiguration, die App lädt immer neu wenn ein Wechsel erkannt wird. Aber von WLAN zu VPN sollte beides lokal sein und kein neu laden verursachen wenn SSID konfiguriert.

**Comment by Uwe1984** [comment]
Genau so ist es :-) ... Ich habe meine WLAN-SSID angegeben und es funktioniert genau so, wie du es beschrieben hast. 
Ich habe jetzt einige Tests mit VPN, mit WLAN gemacht und es läuft super. 

Das einzige was ein bisschen stört, ist die Fehlermeldung bei ersten mal starten der App im Mobil Netz bevor die VPN Verbindung greift.. --> „WebView Error – Die Internetverbindung scheint offline zu sein“ 

Kaffee an dich geht schon mal raus ;-)

**Comment by foxriver76** [comment]
Verschwindet die Fehlermeldung von selbst oder musst du sie aktiv bestätigen?

**Comment by Uwe1984** [comment]
Sie verschwindet von selbst

**Comment by Uwe1984** [comment]
Ich noch ein seltsames Verhalten festgestellt... 
Wenn ich in den Einstellungen bei Extern den "Host IP:PORT wie bei lokal angebe, dann lädt meine Vis zwar ohne dass die VPN Verbindung aufgebaut wird. Allerdings kann ich dann keine Grafana Dashboards mehr öffnen. 

Es kommt ein Popup "Der WEb Adapter benötigt eine Authentifizierung doch Nutzername und/oder Passwort wurden nicht angegeben. 
--> Das ist richtig, ich habe keine Web Logindaten.  

Wenn ich aber in den Einstellungen bei "Extern" keine IP Adresse angebe, kommt das Popup nicht...

**Comment by foxriver76** [comment]
> Sie verschwindet von selbst

Sehr gut, dass sie verschwindet nach erfolgreihem neu laden. Ja wenn dann bräuchte man echt eine Option für extern "nur VPN" wo er dann vorher nichts versucht zu laden und dann halt Ladescreen ist bis ein VPN aufgebaut wurde. Würde auch das verhindern, dass man irgendeinen extern dummy konfigurieren muss.

**Comment by Uwe1984** [comment]
Ich glaube das wäre die sauberere Lösung und würde bei den meiste Usern keine Verwirrung auslösen, wie die Einstellungen zu wählen sind, wenn man nur VPN nutzen möchte

**Comment by foxriver76** [comment]
> Ich noch ein seltsames Verhalten festgestellt... Wenn ich in den Einstellungen bei Extern den "Host IP:PORT wie bei lokal angebe, dann lädt meine Vis zwar ohne dass die VPN Verbindung aufgebaut wird. Allerdings kann ich dann keine Grafana Dashboards mehr öffnen.
> 
> Es kommt ein Popup "Der WEb Adapter benötigt eine Authentifizierung doch Nutzername und/oder Passwort wurden nicht angegeben. --> Das ist richtig, ich habe keine Web Logindaten.
> 
> Wenn ich aber in den Einstellungen bei "Extern" keine IP Adresse angebe, kommt das Popup nicht...
> 
> Edit: Ich bin Zuhause im WLAN

**Comment by Uwe1984** [comment]
Würde aber bedeuten, jedesmal wenn ich  meinen Server neustarte, dann muss ich erstmal die Externe URL wieder löschen. 
Dann bei Grafana einloggen und anschließend wieder bei extern die URL eingeben.

**Comment by foxriver76** [comment]
Also ich fixe das mal, dass er nur bei login bei gleicher ip das macht wie auch bei lokaler. Aber wieso musst du Grafana einloggen? Aber das vermutlich eine Grafana Konfigurations Geschichte. ^^

**Comment by foxriver76** [comment]
Ich mache hier mal zu, für spezielle Issues zu dem Thema bitte ein eigenes Ticket anlegen falls noch was fehlt.
